edx_sga/tasks.py

```diff
@@ -0,0 +1,309 @@
+    """
+    prepare files for download
+
+    Args:
+        course_id (unicode): edx course id
+        block_id (unicode): edx block id
+        location (Location): location of sga module
+        destination_path (str): folder from where we will zip files and facilitate for download
+    """
+    from edx_sga.sga import BLOCK_SIZE
+    submissions_students = SubmissionsStudent.objects.filter(
+        course_id=course_id,
+        item_id=block_id
+    )
+    for submissions_student in submissions_students:
+        submissions = _get_submissions(submissions_student.student_id, course_id, block_id)
+        if submissions:
+            answer = submissions[0]['answer']
+            if answer:
+                student = user_by_anonymous_id(submissions_student.student_id)
+                source_file = six.u('{loc.org}/{loc.course}/{loc.block_type}/{loc.block_id}/{sha1}{ext}').format(
+                    loc=location,
+                    sha1=answer['sha1'],
+                    ext=os.path.splitext(answer['filename'])[1]
+                )
+
+                destination_file = os.path.join(
+                    destination_path,
+                    "{student_name}_{sha1}{ext}".format(
+                        student_name=student.username,
+                        sha1=answer['sha1'],
+                        ext=os.path.splitext(answer['filename'])[1]
+                    )
+                )
+                try:
+                    file_descriptor = default_storage.open(source_file)
+                    with open(destination_file, 'wb') as file_pointer:
+                        while True:
+                            data = file_descriptor.read(BLOCK_SIZE)
+                            if data == '':  # end of file reached
+                                break
+                            file_pointer.write(data)
+
+                except IOError:
+                    log.exception("Unable to download submission of student=%s", student.username)
+        else:
+            continue
+
+
+def _compress_folder(destination_path):
+    """
+    compress given folder/file
+
+    Args:
+        destination_path (str): path (including name) of folder/file which we want to compress.
+    """
+    make_archive(
+        destination_path,
+        'zip',
+        destination_path
+    )
+
+
+@CELERY_APP.task
+def zip_student_submissions(course_id, block_id, location, user_name):
+    """
+    Task to download all submissions as zip file
+
+    Args:
+        course_id (unicode): edx course id
+        block_id (unicode): edx block id
+        location (Location): location of sga module
+        user_name (str): staff user name
+    """
+    submissions_dir_name = get_submissions_dir_name(user_name, block_id, course_id)
+    destination_path = _get_dir_path(submissions_dir_name, location)
+    _mk_dir(destination_path)
+    remove_zip_file(user_name, block_id, course_id)
```

Comment:
`_mk_dir` and `remove_zip_file` are called together because we need to remove an existing directory where the submissions are collected, and the zip file that is directly associated with that directory. It makes more sense to combine these steps into one function called `remove_existing_submission_artifacts` or something like that. That function would remove the directory AND the zip file. We will always want to do both.